src/servers/Server.cc

```diff
@@ -173,10 +173,7 @@ Server::doClientRead(const CommIoCbParams &io)
     // case Comm::COMM_ERROR:
     default: // no other flags should ever occur
         debugs(33, 2, io.conn << ": got flag " << rd.flag << "; " << xstrerr(rd.xerrno));
-        LogTagsErrors lte;
-        lte.timedout = rd.xerrno == ETIMEDOUT;
-        lte.aborted = !lte.timedout; // intentionally true for zero rd.xerrno
-        terminateAll(Error(ERR_CLIENT_GONE, SysErrorDetail::NewIfAny(rd.xerrno)), lte);
+        terminateWithError(ERR_READ_ERROR, rd.xerrno);
```

Comment:
I suggest placing the err_code fix in the PR title while mentioning code deduplication in the PR description. If nothing else, this will confirm that others are not against this arguably controversial err_code change (i.e. we do not want to accidentally sneak in that unwanted change).

Alternatively, consider posting an even smaller official PR dedicated to the  ERR_READ_ERROR fix (leaving deduplication for this followup PR). A smaller PR would be easier to backport if others decide they want to backport this change.

Reply:
> Alternatively, consider posting an even smaller official PR dedicated to the ERR_READ_ERROR fix 

Posted an official PR1720.